src/archivematicaCommon/tests/test_elasticsearch_functions.py

```diff
@@ -88,3 +89,116 @@ def test_set_tags(self):
+            uuid,
+            mets_file_path,
+            index,
+            type_,
+            sipName,
+            identifiers,
+        )
+
+        # ES should have indexed 12 files
+        # - 5 content files
+        # - 5 checksum and csv files in the metadata directory
+        # - 2 files generated in the transfer process
+        assert indexed_files_count == 12
+        assert dummy_try_to_index.call_count == 12
+
+        # Metadata should have been indexed only for these content
+        # files because they are listed in the metadata.csv file
+        content_files_with_metadata = (
+            {
+                'path': ('objects/View_from_lookout_over_Queenstown_'
+                         'towards_the_Remarkables_in_spring.jpg'),
+                'title': ('Morning view from lookout over Queenstown '
+                          'towards the Remarkables in spring'),
+                'creator': 'Pseudopanax at English Wikipedia',
+            },
+            {
+                'path': 'objects/beihai.tif',
+                'title': 'Beihai, Guanxi, China, 1988',
+                'creator': ('NASA/GSFC/METI/ERSDAC/JAROS and U.S./Japan '
+                            'ASTER Science Team'),
+            },
+            {
+                'path': 'objects/bird.mp3',
+                'title': '14000 Caen, France - Bird in my garden',
+                'creator': 'Nicolas Germain',
+            },
+            {
+                'path': 'objects/ocr-image.png',
+                'title': 'OCR image',
+                'creator': 'Tesseract',
+            },
+        )
+        for file_metadata in content_files_with_metadata:
+            dc = indexed_data[file_metadata['path']]
+            assert dc['dc:title'] == file_metadata['title']
+            assert dc['dc:creator'] == file_metadata['creator']
+
+        # There is no metadata for this content file because
+        # it was not listed in the metadata.csv file
+        assert indexed_data['objects/piiTestDataCreditCardNumbers.txt'] is None
+
+        # Checksum and csv files in the metadata directory
+        # won't have dublin core metadata indexed
+        files_in_metadata_directory = (
+            'checksum.md5',
+            'checksum.sha1',
+            'checksum.sha256',
+            'metadata.csv',
+            'rights.csv',
+        )
+        for filename in files_in_metadata_directory:
+            path = 'objects/metadata/transfers/DemoTransfer-{}/{}'.format(
+                mets_object_id, filename
+            )
+            assert indexed_data[path] is None
+
+        # Neither will the generated files during the transfer process
+        generated_files = (
+            'dc.json',
+            'directory_tree.txt',
+        )
+        for filename in generated_files:
+            path = 'objects/metadata/transfers/DemoTransfer-{}/{}'.format(
+                mets_object_id, filename
+            )
+            assert indexed_data[path] is None
```

Comment:
Great test! It will require some changes after the ES upgrade work is merged but it shouldn't be hard.